jenkins_jobs/modules/hipchat_notif.py

```diff
@@ -131,12 +133,26 @@ def gen_xml(self, xml_parent, data):
                 logger.warning("'room' is deprecated, please use 'rooms'")
                 hipchat['rooms'] = [hipchat['room']]
 
-        properties = xml_parent.find('properties')
-        if properties is None:
-            properties = XML.SubElement(xml_parent, 'properties')
-        pdefhip = XML.SubElement(properties,
-                                 'jenkins.plugins.hipchat.'
-                                 'HipChatNotifier_-HipChatJobProperty')
+        plugin_info = self.registry.get_plugin_info("Jenkins HipChat Plugin")
+        version = pkg_resources.parse_version(plugin_info.get('version', '0'))
+
+        if version >= pkg_resources.parse_version("0.1.9"):
+            publishers = xml_parent.find('publishers')
+            if publishers is None:
+                publishers = XML.SubElement(xml_parent, 'publishers')
+
+            logger.warning(
+                "'hipchat' module supports the old plugin versions <1.9, "
+                "newer versions are supported via the 'publishers' module. "
+                "Please upgrade you job definition")
+            return hipchat_pub(self.registry, publishers, hipchat)
```

Comment:
Tests are passing now (`tox -e py27,py34,pep8`), thanks!

For me, the tests are working with the "registry.dispatch" version too. So what does not work for you specifically? Passing tests ensure that the transformation from yaml to xml works correctly and there's not much other room for breakage..